File: runme/mcgen/local/datan/exasrc/e1gam.f
      PROGRAM E1GAM
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
C identify program to user
      WRITE(*,*)' Program E1GAM demonstrates use of'
      WRITE(*,*)' GGAMMA and GLNGAM'
      WRITE(*,*)' and provides interactively numerical values'
      WRITE(*,*)' of the gamma function and its logarithm'
      WRITE(*,*)' '
      WRITE(*,*)' Choose'
      WRITE(*,*)' 1 - gamma function'
      WRITE(*,*)' 2 - log of gamma function'
      WRITE(*,*)' >'
      READ(*,*)ISWIT
      WRITE(*,*)' Enter x'
      WRITE(*,*)' >'
      READ(*,*)X
      IF(ISWIT.EQ.1) THEN
C gamma
        F=GGAMMA(X)
        WRITE(*,*)' gamma(x) = ',F
      ELSE IF(ISWIT.EQ.2) THEN
C log gamma
        F=GLNGAM(X)
        WRITE(*,*)' log (gamma(x)) = ',F
      END IF
      END

File: runme/mcgen/local/datan/exasrc/e1gam.c
#include <stdio.h>
#include "datsrc.h"

main(void)
{
    static integer iswit;
    static double f, x;
/* identify program to user */
    printf("%s\n", "Program E1GAM demonstrates use of");
    printf("%s\n", "GGAMMA and GLNGAM");
    printf("%s\n", "and provides interactively numerical values");
    printf("%s\n\n", "of the gamma function and its logarithm.");
    printf("%s\n", "Choose");
    printf("%s\n", "1 - gamma function");
    printf("%s\n", "2 - log of gamma function");
    printf("%s", "> ");
    scanf("%li", &iswit);
    printf("%s\n", "Enter x:");
    printf("%s", "> ");
    scanf("%lf", &x);
    if (iswit == 1) {
/* gamma */
	f = ggamma_(&x);
        printf("%s%14.10lf\n", "gamma(x) = ", f);
    } else if (iswit == 2) {
/* log gamma */
	f = glngam_(&x);
        printf("%s%14.10lf\n", "log (gamma(x)) = ", f);
    }
    return 0;
} /* main */

